libobs/obs-scene.c

```diff
@@ -2082,6 +2084,56 @@ void obs_sceneitem_set_order_position(obs_sceneitem_t *item, int position)
 	obs_scene_release(scene);
 }
 
+void obs_scene_set_items_order(obs_scene_t *scene, int64_t* new_items_order, int items_count)
+{
+	if (!scene || items_count <= 1)
+		return;
+
+	obs_scene_addref(scene);
+	full_lock(scene);
+
+	//create array with items of this scene
+	obs_sceneitem_t **scene_items_cached = bzalloc(items_count * sizeof(obs_sceneitem_t*)); 
+	int index = 0;
+	scene_items_cached[0] = scene->first_item;
+	while (scene_items_cached[index] && index < items_count - 1) {
+		scene_items_cached[index+1] = scene_items_cached[index]->next;
+		index++;
+	}
+
+	if (scene_items_cached[items_count-1] == NULL || scene_items_cached[items_count-1]->next != NULL) {
+		blog(LOG_ERROR, "obs_scene_set_items_order: Wrong items count in order array");
+	} else {
+		//deattach all items from scene 
+		index = 0;
+		while (scene_items_cached[index] && index < items_count - 1) {
+			detach_sceneitem(scene_items_cached[index]);
```

Comment:
Can this be part for the loop above? Meaning we cache the item and we detach it or does it have to be in 2 separate loops?

Reply:
Outdated but still relevant as if a second loop is needed here

Reply:
It will made code complex in case of something wrong with number of items and we need to restore list from not correctly filled array.  and it will not give much speed up.

Reply:
Got it, makes sense.